pkg/registry/apis/query/parser.go

```diff
@@ -80,7 +80,7 @@ func (p *queryParser) parseRequest(ctx context.Context, input *query.QueryDataRe
 			return rsp, MakePublicQueryError(q.RefID, "multiple queries with same refId")
 		}
 
-		ds, err := p.getValidDataSourceRef(ctx, q.Datasource, q.DatasourceID)
+		ds, err := p.getValidDataSourceRef(ctx, q)
```

Comment:
I like this refactoring. It feels like we're removing the "Data clump" code smell by making this refactoring.
But I would have liked to see the refactoring as its own separate commit - I think it would have made this PR easier to review.

Reply:
ah my bad, I needed to pass the entire `DataQuery` struct to call `.Get("datasourceUid")` later so I kept it together so the commit could "work on its own".